erpnext/accounts/report/financial_statements.py

```diff
@@ -167,53 +300,99 @@ def get_appropriate_currency(company, filters=None):
-					raise_exception=1
-				)
+					raise_exception=1)
+					
+			#generate accumulated data for the different periods specified by the user
 			for period in period_list:
-				# check if posting date is within the period
 
+				# check if posting date is within the period
 				if entry.posting_date <= period.to_date:
+					
+					#we ignore accumulated values for reports like P\L and cash flow
 					if (accumulated_values or entry.posting_date >= period.from_date) and \
 						(not ignore_accumulated_values_for_fy or
-							entry.fiscal_year == period.to_date_fiscal_year):
+							entry.fiscal_year == period.fiscal_year_end_date):
+
+						#period.key updates the account value for the given period (Week/Month/Year)
 						d[period.key] = d.get(period.key, 0.0) + flt(entry.debit) - flt(entry.credit)
 
 			if entry.posting_date < period_list[0].year_start_date:
 				d["opening_balance"] = d.get("opening_balance", 0.0) + flt(entry.debit) - flt(entry.credit)
 
 
 def accumulate_values_into_parents(accounts, accounts_by_name, period_list, accumulated_values):
-	"""accumulate children's values in parent accounts"""
+	"""
+	Accumulate children's values in parent accounts
+
+	Args:
+		accounts (list): parent/child account names arranged by depth
+		accounts_by_name (dict): account details mapped by account_name
+		period_list (list of dict): time frames across which the data is needed
+		accumulated_values (int): whether values are accumulated or not
+	"""	
 	for d in reversed(accounts):
 		if d.parent_account:
 			for period in period_list:
 				accounts_by_name[d.parent_account][period.key] = \
 					accounts_by_name[d.parent_account].get(period.key, 0.0) + d.get(period.key, 0.0)
-
+			#accumulate the balance in the parent account's opening balance variable
 			accounts_by_name[d.parent_account]["opening_balance"] = \
 				accounts_by_name[d.parent_account].get("opening_balance", 0.0) + d.get("opening_balance", 0.0)
 
 
 def prepare_data(accounts, balance_must_be, period_list, company_currency):
+	"""
+	Creates each row as a dictionary that is finally rendered for the user 
+
+	Args:
+		accounts (list of dict): parent/child account names arranged by depth
+		balance_must_be (string): debit or credit
+		period_list (list of dict): list of periods that need to processed
+		company_currency (string): currency
+
+	Returns:
+		list of dict: each of the rows that need to be rendered 
+	"""
+	#take the starting and ending date
 	data = []
 	year_start_date = period_list[0]["year_start_date"].strftime("%Y-%m-%d")
 	year_end_date = period_list[-1]["year_end_date"].strftime("%Y-%m-%d")
-
+	
+	#loop through all the accounts and create rows for each with neccessary data entries
 	for d in accounts:
-		# add to output
 		has_value = False
 		total = 0
 		row = frappe._dict({
 			"account": _(d.name),
 			"parent_account": _(d.parent_account) if d.parent_account else '',
+			"lft": _(d.lft), 
+			"rgt": _(d.rgt),
```

Comment:
```suggestion
			"lft": d.lft, 
			"rgt": d.rgt,
```
- Do not translate `lft` and `rgt` values, will mess up the tree, since they arent supposed to be translated.
- They are internally used by system and is never visible to the user